streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -4278,6 +4286,13 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
       ZStream.fromEffect(ZIO.serviceWith(f))
   }
 
+  final class ServiceWithStreamPartiallyApplied[Service](private val dummy: Boolean = true) extends AnyVal {
+    def apply[E, A](f: Service => ZStream[Has[Service], E, A])(implicit
```

Comment:
I am not sure about this. If the user is using our pattern, why would they need this?

```scala
for {
  serviceA <- ZStream.service[ServiceA]
  serviceB <- f(serviceA)
} yield serviceB
```

Reply:
It the counterpart of the `serviceWith` operator. So I think it comes up in two cases.

First, implementing service accessors:

```scala
trait Persistence {
  def loadOrders(customerID: CustomerId, from: OrderId, to: OrderId): ZStream[Any, Nothing, Customer]
}

object Persistence {
  def loadOrders(customerID: CustomerId, from: OrderId, to: OrderId): ZStream[Has[Persistence], Nothing, Customer] =
    ZStream.serviceWithStream(_.loadOrders(customerId, from, to))
}
```

Second, if you are working with a service in normal application code and not in a service implementation, for example in test code, it can be useful to access a service in the environment do something with it that returns a stream.

We have `serviceWith` and `serviceWithManaged` so I think either we should have this for symmetry or we should revisit those operators. I agree all of these accessors are becoming less important as we access services more within other services but still seems like they are useful in some cases.